### DISCUSSION TITLE
How to Auto-Import a Server Function from a Nuxt Module?

### DISCUSSION THREAD

**Discussion by rickgregg** [nuxt/nuxt#27226] [Questions] [answered]
How do you auto import a server function from a NUXT module into a Nuxt Server Application? Tried the following in **module.ts** without success.  I can auto-import into client side code all day long. What's the technique? Thank you.
```
// server library
addImportsDir(resolver.resolve("./runtime/server/utils"))
```

**Comment by rickgregg** [comment, marked as answer]
I found a composable in Nuxt Kit **addServerImportsDir** that did the trick. May want to consider updating the Module Recipes Section on the Nuxt Website with an example. Thanks.

**Reply by TheAlexLichter** [reply]
@rickgregg PR welcome 🔥

**Comment by 906212839** [comment]
public directory after build.output is dynamically loaded how to let him access the file. I found that the asset variable in server stored the public before packaging what should I do that can be dynamically added to the output/public file can be accessed correctly?
